editor/src/clj/editor/atlas.clj

```diff
@@ -217,7 +217,12 @@
   (inherits outline/OutlineNode)
 
   (property id g/Str
-            (value (g/fnk [maybe-image-resource] (some-> maybe-image-resource resource/proj-path path->id)))
+            (value (g/fnk [maybe-image-resource rename-patterns] 
+                          (let [rename-patterns (if (nil? rename-patterns) "" rename-patterns)
+                                id (some-> maybe-image-resource resource/proj-path path->id)]
+                            (if (nil? id)
+                              ""
+                              (AtlasUtil/replaceStrings rename-patterns id)))))
```

Comment:
At first, I wrote a nice Clojure thing, but then also opted out of it, since I didn't want to have the logic in two places.

Reply:
I think it is good that we share the code. A potential improvement would be to only do the tokenization & preparations once in the `AtlasNode` whenever the `rename-patterns` property changes so we don't have to repeat it for every image. But I think we can do that at a later stage.